mock/py/mockbuild/buildroot.py

```diff
@@ -216,7 +216,11 @@ def _init(self, prebuild, do_log):
                     and self.config['clean']):
                 update_state = "{0} update".format(self.pkg_manager.name)
                 self.state.start(update_state)
-                self.pkg_manager.update()
+                output = self.pkg_manager.update()
+                if ("lvm_root" in self.plugins.plugins and
+                        "Upgraded:" in output):
+                    self.plugins.call_hooks('recreate_snapshot')
+                    self.bootstrap_buildroot.plugins.call_hooks('recreate_snapshot')
```

Comment:
Thanks for the patch.  There's `all_chroot_packages` method somewhere in mock code, using that would be much more reliable way to detect something changed than parsing the dnf output.

Reply:
Calling the hook should be plugin agnostic, simply "if something changed => call the hook".  Ie don't ask for `lvm_root` plugin here.

Reply:
This makes sense, thanks for the hint.